jupyter-js-widgets/src/widget.ts

```diff
@@ -391,31 +391,34 @@ class WidgetModel extends Backbone.Model {
     /**
      * Serialize widget state.
      *
-     * A serializer is a function which takes in a state attribute and a widget,
-     * and synchronously returns a JSONable object. The returned object will
-     * have toJSON called if possible, and the final result should be a
-     * primitive object that is a snapshot of the widget state that may have
-     * binary array buffers.
+     * @param state - a dictionary of state to be serialized
+     *
+     * A serializer is a function which takes in a state attribute and a Widget
+     * instance, and synchronously returns a JSONable object. The state values
+     * of the returned object will have toJSON called if possible, and the final
+     * result should be a primitive object that is a snapshot of the widget
+     * state that may have binary array buffers.
      */
     serialize(state) {
         const serializers = (this.constructor as typeof WidgetModel).serializers || {};
-        for (const k of state) {
+        let newState = {};
+        Object.keys(state).forEach(k => {
             try {
                 if (serializers[k] && serializers[k].serialize) {
-                    state[k] = (serializers[k].serialize)(state[k], this);
+                    newState[k] = (serializers[k].serialize)(state[k], this);
                 } else {
                     // the default serializer just deep-copies the object
-                    state[k] = JSON.parse(JSON.stringify(state[k]));
+                    newState[k] = JSON.parse(JSON.stringify(state[k]));
                 }
-                if (state[k].toJSON) {
-                    state[k] = state[k].toJSON();
+                if (newState[k].toJSON) {
+                    newState[k] = newState[k].toJSON();
                 }
             } catch (e) {
-                console.error("Error serializing widget state attribute: ", k);
+                console.error(`Error serializing widget state attribute ${k} (value ${state[k]})`);
                 throw e;
```

Comment:
I believe IE doesnt support this, right @SylvainCorlay, what's it called, template strings orso?